### PR TITLE
Make adequacy patch parameters future-proof

### LINKED ISSUE #762: Adequacy patch parameters in wrong section
TODO : move parameters to section **adequacy patch**, rename some to match RTE-i's naming conventions.

### PR DESCRIPTION
**Pull request by flomnes**
Move adequacy-patch parameters to section **adequacy patch** (current : **optimization**). This is to match RTE-i's development branch.

### CODE REVIEW

File: src/libs/antares/study/parameters.cpp [hunk 1]
```diff
@@ -623,6 +624,22 @@ static bool SGDIntLoadFamily_Optimization(Parameters& d,
     }
     return false;
 }
+static bool SGDIntLoadFamily_AdqPatch(Parameters& d,
+                                      const String& key,
+                                      const String& value,
+                                      const String&,
+                                      uint)
```
Comment (guilpier-code) [inline src/libs/antares/study/parameters.cpp:631]: For information, the last argument of **every SGDIntLoadFamily_some_section(..., uint)** is useless. We should remove it for every function.
Either we correct it in this PR or we create a ticket. Correcting it now would be very quick.

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/parameters.cpp (hunk 1 above)
Comment (flomnes) [reply src/libs/antares/study/parameters.cpp:631]: The legacy function uses this argument as a study version number.

https://github.com/AntaresSimulatorTeam/Antares_Simulator/blob/eacccf28ca43f2d714365c1da73a41ee4136c57e/src/libs/antares/study/parameters.cpp#L923-L927

Comment (guilpier-code) [reply src/libs/antares/study/parameters.cpp:631]: ok. 
here I correct my previous remark.
Note that this new remark leads to a possible refactor. Let's discuss if we need to take this remark into account.

Every **Callback** function that is a value of the map **sectionAssociatedToKeysProcess** do not need the last **uint version** argument. the **Callback** type should also loose the unnecessary argument.
So **SGDIntLoadFamily_Legacy** is kind of apart from the  **SGDIntLoadFamily**.

Another point : the 4th argument **const String&** (which the raw value, before the value is turned to lower case) is only used by the body of **SGDIntLoadFamily_General**. So there is a kind of repetition in the list of arguments of this family of function, example : 
```C++
handleAllKeysInSection(*this, p->key, value, p->value, version)
```
We could try to remove one of the **value** argument as well.

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/parameters.cpp (hunk 1 above)
Comment (flomnes) [reply src/libs/antares/study/parameters.cpp:631]: See #774

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**